### DISCUSSION TITLE
middleware('auth') on non-Filemant-routes

### DISCUSSION THREAD

**Discussion by wpas81** [filamentphp/filament#10788] [Help]
Hi

can I use the Filament login functionality for non-filemant pages? I'd like to have a few other pages behind a login-screen. If I add ->middleware('auth') to the route it returns 'no route for login found'. 

I could add my own logic or implement JetStream but it's already  built into Filament so it seems it should be possible no?

thanks in advance

**Comment by wpas81** [comment]
anyone?

**Reply by MNhat0308** [reply]
same i need this too!

**Comment by helloreece** [comment]
_**Fair warning: I'm a new developer. Check anything I suggest against the official documentation. **_ 

Your approach seems fine. Filament's login authenticates a user using a guard ('web' by default). Applying the auth middleware to non-Filament routes checks that the user is authenticated (again, using 'web' guard by default). If they logged in via Filament's login, they will pass this test and follow the route. If not, they are redirected to the 'login' route.

So your error is appearing because the 'login' route within auth middleware doesn't exist.

> class Authenticate extends Middleware
> {
>     /**
>      * Get the path the user should be redirected to when they are not authenticated.
>      */
>     protected function redirectTo(Request $request): ?string
>     {
>         return $request->expectsJson() ? null : route('login');
>     }
> }

Here's what I'd do:

1. Have you got ->login() in your Filament config? If not, add it in and see what the login route is that is added.
2. It's likely that the login route being registered isn't 'login' but instead is panel specific (e.g. 'filament.admin.auth.login' or similar). 
3. You could write your own 'login' route that maps to the Filament panel's login route.
4. You could revise the auth middleware to redirect to your Filament admin's login route. Example below.

> class Authenticate extends Middleware
> {
>     /**
>      * Get the path the user should be redirected to when they are not authenticated.
>      */
>     protected function redirectTo(Request $request): ?string
>     {
>         return $request->expectsJson() ? null : route('filament.admin.auth.login');
>     }
> }

**Comment by MansourM** [comment]
look here `https://github.com/filamentphp/filament/discussions/5226`

**Comment by dallin** [comment]
If you want to protect a specific route in `web.php` behind Filament's auth middleware, you have two options as far as I can tell:

Directly specify the Filament middleware:
```php
Route::middleware(\Filament\Http\Middleware\Authenticate::class)->get('blahblahblah', BlahController::class);
```
**OR**

Override the default `auth` middleware alias if you don't need it:
```php
// bootstrap/app.php

    ->withMiddleware(function (Middleware $middleware) {
        $middleware->alias([
            'auth' => \Filament\Http\Middleware\Authenticate::class,
        ]);
    })
```
